### PR TITLE
Adding an isUserAdmin function for windows

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

File: osquery/core/windows/process_ops.cpp [hunk 1]
```diff
@@ -70,4 +71,23 @@ boost::optional<std::string> getEnvVar(const std::string &name) {
 void cleanupDefunctProcesses() {}
 
 void setToBackgroundPriority() {}
+
+// Helper function to determine if thread is running with admin privilege.
+bool isUserAdmin() {
```
Comment (PoppySeedPlehzr) [inline osquery/core/windows/process_ops.cpp]: This function checks if the current thread's token has elevated privs. Is this function name the most appropriate? Should it be something else? `runningAsAdmin`?

File: include/osquery/system.h [hunk 2]
```diff
@@ -289,5 +291,11 @@ Status createPidFile();
 // Notice that they are swapped when compared to POSIX FUNCTION_r.
 struct tm* gmtime_r(time_t* t, struct tm* result);
 struct tm* localtime_r(time_t* t, struct tm* result);
+/**
+ * @brief Windows getter for determining Admin status
+ *
+ * @return A bool indicating if the current process is running as admin
+ */
+bool isUserAdmin();
```
Comment (theopolis) [inline include/osquery/system.h]: Can we get this moved outside of the `WINDOW` compiler gate? And then on POSIX it's just returns `getuid() == 0`?

**Comment by osqueryer** [comment]
:no_good: The commit c5f6d1d696e3f1540d76895ca9ff1a2c477c6888 (Job results: [159](https://jenkins.osquery.io/job/osqueryPullRequestAudit/159/)) failed the code audit.

File: osquery/core/windows/process_ops.cpp [hunk 3]
```diff
@@ -70,4 +71,27 @@ boost::optional<std::string> getEnvVar(const std::string &name) {
...
+  HANDLE hToken = nullptr;
+  if (!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &hToken)) {
+    return false;
+  }
+  TOKEN_ELEVATION Elevation;
+  DWORD cbSize = sizeof(TOKEN_ELEVATION);
+  if (GetTokenInformation(
+          hToken, TokenElevation, &Elevation, sizeof(Elevation), &cbSize) ==
+      0) {
+    return false;
```
Comment (theopolis) [inline osquery/core/windows/process_ops.cpp]: Will you leak the token here since the `HANDLE` is not closed?

**Comment by osqueryer** [comment]
:no_good: The commit b48b6d03d95173e2b5a8df4472d39b9ada2bc62c (Job results: [164](https://jenkins.osquery.io/job/osqueryPullRequestAudit/164/)) failed the code audit.

**Comment by theopolis** [comment]
What happened with this PR? Why does Github say I added a commit? The squash merge seems to have recorded both PRs in the resultant commit too.

**Comment by PoppySeedPlehzr** [comment]
Yeah that was my bad. I'll fix this :3

**Pull request merged**